pkg/kubeadm/cmd/cmd.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"io"
+
+	"github.com/renstrom/dedent"
+	"github.com/spf13/cobra"
+
+	kubeadmapi "k8s.io/kubernetes/pkg/kubeadm/api"
+)
+
+func NewKubeadmCommand(in io.Reader, out, err io.Writer, envParams map[string]string) *cobra.Command {
+	cmds := &cobra.Command{
+		Use:   "kubeadm",
+		Short: "kubeadm: bootstrap a secure kubernetes cluster easily.",
+		Long: dedent.Dedent(`
+			kubeadm: bootstrap a secure kubernetes cluster easily.
+
+			    ┌──────────────────────────────────────────────────────────┐
+			    │ KUBEADM IS ALPHA, DO NOT USE IT FOR PRODUCTION CLUSTERS! │
+			    │                                                          │
+			    │ But, please try it out! Give us feedback at:             │
+			    │ https://github.com/kubernetes/kubernetes/issues          │
+			    │ and at-mention @kubernetes/sig-cluster-lifecycle         │
+			    └──────────────────────────────────────────────────────────┘
+
+			Example usage:
+
+			    Create a two-machine cluster with one master (which controls the cluster),
+			    and one node (where workloads, like pods and replica sets run).
+
+			    ┌──────────────────────────────────────────────────────────┐
+			    │  On the first machine                                    │
+			    ├──────────────────────────────────────────────────────────┤
+			    │ master# kubeadm init master                              │
```

Comment:
Sucks indeed, and let's keep this for later time.

On Wed, 31 Aug 2016, 22:42 Joe Beda, <notifications@github.com> wrote:

> In pkg/kubeadm/cmd/cmd.go
> <https://github.com/kubernetes/kubernetes/pull/31221#discussion_r77079266>
> :
>
> > +			    │ KUBEADM IS ALPHA, DO NOT USE IT FOR PRODUCTION CLUSTERS! │
> > +			    │                                                          │
> > +			    │ But, please try it out! Give us feedback at:             │
> > +			    │ https://github.com/kubernetes/kubernetes/issues          │
> > +			    │ and at-mention @kubernetes/sig-cluster-lifecycle         │
> > +			    └──────────────────────────────────────────────────────────┘
> > +
> > +			Example usage:
> > +
> > +			    Create a two-machine cluster with one master (which controls the cluster),
> > +			    and one node (where workloads, like pods and replica sets run).
> > +
> > +			    ┌──────────────────────────────────────────────────────────┐
> > +			    │  On the first machine                                    │
> > +			    ├──────────────────────────────────────────────────────────┤
> > +			    │ master# kubeadm init master                              │
>
> It sucks that we need to run as root. It would be nice to be able to have
> some way to direct the kubelet without running as root. If we start self
> hosting the master components (and have an admin channel to the kubelet) we
> can get rid of the need to run as root. Probably worth an issue/TODO at
> this point.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31221/files/57096ca902c5c666d45df191037d9278972af1fd#r77079266>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPWS21OCeIS1KI92EjRtuJcB3Ce2IiWks5qlfVdgaJpZM4JqsTH>
> .
>